pkgs/development/libraries/libspnav/configure-socket-path.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/spnav.c b/spnav.c
+index f9e10f8..f9a117b 100644
+--- a/spnav.c
++++ b/spnav.c
+@@ -36,7 +36,8 @@ OF SUCH DAMAGE.
+ #include <sys/select.h>
+ #include "spnav.h"
+ 
+-#define SPNAV_SOCK_PATH "/var/run/spnav.sock"
++#define DEFAULT_SPNAV_SOCK_PATH "/var/run/spnav.sock"
++#define SPNAV_SOCK_PATH_ENV "SPNAVD_SOCK_LOCATION"
```

Comment:
I wouldn't go with having to invoke applications with an environment variable, but just prope `$XDG_RUNTIME_DIR/spnav.sock` if `$XDG_RUNTIME_DIR` is set (which is the case in a user session), and then fall back to `/run/spnav.sock`.